pkg/controller/podautoscaler/horizontal.go

```diff
@@ -1177,16 +1177,22 @@ func setConditionInList(inputList []autoscalingv2.HorizontalPodAutoscalerConditi
 	return resList
 }
 
-func max(a, b int32) int32 {
-	if a >= b {
-		return a
+func max(proposals []int32) int32 {
+	maxValue := proposals[0]
```

Comment:
If the length of policy is 0, the result would cause an error

Reply:
ScalingRules have default values
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/v2beta2/defaults.go#L33